sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/PartitionRecordReader.scala

```diff
@@ -29,13 +29,3 @@ class PartitionRecordReader[T](
 
   override def close(): Unit = rowReader.close()
 }
-
-class PartitionRecordReaderWithProject[X, T](
```

Comment:
can we make this class to append partition values?

I feel it's weird to have these 3 methods in `FilePartitionReaderFactory`: `buildReader`, `buildReaderWithPartitionValues` and `buildColumnarReaderWithPartitionValues`. They are not symmetrical.

Reply:
Which class do you mean? I think it is bit complex to refactor the current `ColumnarBatch` to support appending partition values. I should create another JIRA for it. What do you think?

Reply:
it's just different ways to abstract the code: 1) add common methods in the parent class 2) create util classes.

In this case I feel 2) is better, because the added methods in the parent class are not symmetrical.